### PR TITLE
Fix build, cleanup, use intersphinx and add GHA workflow

### PR DESCRIPTION
**Pull request by umarcor**
The build of this repository seems to be broken since august. This PR brings several updates and cleanups to have it alive again.

- A GitHub Actions workflow (`Doc`) is added to allow contributors to test changes without having to setup a new RTD project. Builds are pushed to `gh-pages`. It is up to each namespace to activate GitHub Pages.
- The main reason for failing builds is the usage of outdated symbolator and hdlparse, which are not compatible with recent versions of Python. Those are replaced with the forks in org 'hdl', which are updated to work with Python 3 only.
- Since GitHub provides tarballs/zipfiles for each branch, instead of requiring git to retrieve dependencies through pip, tarballs/zipfiles are used.
- The main root of complexity in this repo is adding extra content through git submodules, which requires supporting all the extensions and tweaks required by each submodule. Instead, this PR uses intersphinx. Each project should be built on its own and published to the desired site. Then, intersphinx allows cross-referencing any item in the inventory.
  - Note that intersphinx does not work in toctree items. In those cases, full URLs are used and a unicode symbol is added.
  - The configuration, the requirements file, the environment, gitignore and the makefile are cleaned up accordingly.
  - Since prjxray is not built in this repo anymore, subdir doxygen is removed.
  - ~Due to the simplification of the structure, the content of `source` is moved to the root of the repo.~

Compare the current and the proposed builds:

- https://symbiflow.readthedocs.io/en/latest/
- https://umarcor.github.io/symbiflow-docs/

### CODE REVIEW

File: index.rst [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+.. toctree::
+   :maxdepth: 3
+
+   introduction
+   toolchain-desc/index
+   Architecture Definitions ➚ <https://symbiflow.readthedocs.io/projects/arch-defs/en/latest/>
+   Project X-Ray ➚ <https://symbiflow.readthedocs.io/projects/prjxray/en/latest/>
+   Project Trellis ➚ <https://prjtrellis.readthedocs.io/en/latest/>
+   FPGPA Assembly (FASM) ➚ <https://fasm.readthedocs.io/en/latest/>
```
Comment (tmichalak) [inline index.rst:13]: FPGPA -> FPGA

**Review by tmichalak — COMMENTED** [review]
In overall I like this cleanup effort. 
I believe that the only concern is about what happens if all the other projects change their theme to a different one than in symbiflow-docs. However, I guess it's up to us to prevent such situation from happening. 
Also, I see that this doc is generated once a week. What about the other projects? I guess we should adjust the cron schedule for those as well.

**Comment by tmichalak** [comment]
@kgugala @mgielda

**Comment by umarcor** [comment]
> I believe that the only concern is about what happens if all the other projects change their theme to a different one than in symbiflow-docs.

Strictly, that is not a problem. It's irrelevant from intersphinx's point of view. Currently, symbiflow-docs, arch-defs, prjxray, prjtrellis and FASM are all using the same (or very similar theme). But that is just "a coincidence". VPR is using a different theme, and still it works the same.

Overall, Sphinx generates an "inventory" file. You can see it with e.g. `python -m sphinx.ext.intersphinx https://docs.python.org/3/objects.inv` (find further details in https://www.sphinx-doc.org/en/master/usage/extensions/intersphinx.html). The content of the inventory is related to the rst sources, toctrees, titles, etc. but unrelated to the theme.

So, for instance, the prjtrellis fork in SymbiFlow org might use a different theme from the one in YosysHQ, each published to a different location. Still, the inventory file will look the same. Naturally, we need to decide if we want to cross-ref to one place or the other.

> However, I guess it's up to us to prevent such situation from happening.

I think we want to keep some homogeneous style in all the repos belonging strictly to SymbiFlow. Therefore, in that regard, it's up to us to take care of it. However, I believe we do also want to have cross-references to the tools with independent Sphinx documentation maintained by someone else. I'm thinking about e.g. VTR, GHDL, Surelog, hdl/containers, hdl/constraints, EDAA,... In those cases, I believe it is desirable indeed for the themes to be different, since it makes more explicit for users that SymbiFlow is built by integrating multiple tools from several communities.

> Also, I see that this doc is generated once a week. What about the other projects? I guess we should adjust the cron schedule for those as well.

On the one hand, the weekly cron is absolutely arbitrary (that's what I'm using in hdl/containers, where I copied it from). We can set to the frequency you want (daily would be ok I think). On the other hand, yes, I do want to apply similar cleanups/updates to (at least) arch-defs, examples, fasm and prjxray. This first PR is to get to know how you feel about the approach overall.

**Pull request merged**